Interview/demo.md

```diff
@@ -22,62 +22,74 @@ Before proceeding to the section of your choice, please consider the following t
 
 * We like code that is simple, but [simple is different from easy](https://www.infoq.com/presentations/Simple-Made-Easy).
 * Keep in mind the [SOLID principles](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) when doing the project.
-* We left out of the requirements whether or not you should try to download everything (posts + each post detail). This is up to you to decide and to justify.
+* We left out of the requirements whether or not you should try to download everything (photos + each photo detail). This is up to you to decide and to justify.
 * Testing is very important for us. Even if you don't write a single test (for instance, because of time constraints), your app should be easy to test (we love [Dependency injection](https://en.wikipedia.org/wiki/Dependency_injection)).
 * Error scenarios should be taken into consideration and it should be easy to add them, even if you don't explicitly handle them (e.g. using an `UIAlertController`).
 * Although UI and UX are important, we are more concerned in this demo with your thought process and with how you architect your application. Your demo should take into consideration features that might be added in the future.
-* You can use any 3rd party libraries you wish (`Alamofire`, `ReactiveCocoa`, `PromiseKit`, `Realm`, etc) but be prepared to justify why you did so. Feel free to use package managers to handle them.
+* You can use any 3rd party libraries (except for networking) if you wish ( `ReactiveCocoa`, `PromiseKit`, `Realm`, etc) but be prepared to justify why you did so. Feel free to use package managers to handle them.
```

Comment:
> but be prepared to justify why you did so

Considering this why do we need to exclude networking? Realm is fine but using a network library is not? Because persistence is also one of the requirements of the project so I don't see why we are ok in allowing Realm but then we don't allow anything for networking even if the candidate knows the library quite well and he/she is able to explain what it's happening behind the scenes. Would be great to have a justification about the reasoning for this direction and what's the main benefit, not only for us but for the candidates too.

Reply:
I agree with @dmcrodrigues. Why exclude 3rd party networking libs? It's not as if we do any network coding anyway.

Reply:
The networking API provided by Apple is powerful and easy enough to setup. It is easy to meet the needs of our Test as well with that simple setup.  Core data on the other hand, which is the Apple preferred mode of handling persistence is not exactly known for easy setup especially for a small tech test. There are other libraries which vastly reduce effort in implementing various aspects - could be powerful animation, automatic diffing etc. This is why we went with that exception. Again if the team feels strongly against it, we can remove the exception. We were just looking to expand scope of test in simple ways to allow candidate some space to exhibit more technical knowledge

Reply:
I personally disagree with the option taken and I don't buy the reasoning behind it, we can always extend the same thought to Core Data, the test is simple enough that shouldn't be too much of a hassle to use Core Data to store the images or anything needed, the APIs have been improving in the last couple of years, Xcode generates a stack for you, ...

To me feels like we are restricting something based on a personal conviction, using a 3rd party project doesn't necessarily mean the candidate doesn't know about the native APIs **but he/she may decide to use a different project to avoid re-inventing the wheel all over again** or are we expecting to have URLs being created directly without any kind of abstraction? We have our own `Resource` type and a couple of abstractions, for a new project something like Moya, as an example, could save the candidate from doing everything from scratch at least for networking, for the rest we say it's ok as long as you are able to justify the usage. To me sounds a bit incoherent and, as I said, there's nothing in the description that explains why we don't allow any 3rd party dependency just for networking.

Reply:
Fair enough, lets get rid of this then. @sergdort  and @konrad-em thoughts?

Reply:
What we thought is that the test requires to make simple GET network calls. But on the other hand implementing networking without 3rd party lib give an opportunity for a candidate to show have they would architect certain aspects of Networking, which spawns more interesting conversations during interview.

Reply:
Sometimes candidates use 3rd party libs for everything and interview becomes very boring.

Reply:
I understand the reasoning behind this choice. However, IMO I don't think we should have this restriction. We should allow the user to use the libraries they prefer and then we can question that choice in the interview. It can also be interesting to see how the candidate can isolate their code from the library.

Regarding the interview, it is our job to make the interview engaging and interesting regardless of the technical choices of the candidate.